File: BE/MoviesAPI/Dtos/Movies/GetMovieDto.cs
using Microsoft.EntityFrameworkCore.Query.SqlExpressions;
using MoviesAPI.Dtos.Categories;
using MoviesAPI.Dtos.MovieTags;
using MoviesAPI.Dtos.VotedMovies;

namespace MoviesAPI.Dtos.Movies
{
    public class GetMovieDto
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string PosterImageUrl { get; set; }
        public int Views { get; set; }
        public int Upvotes { get; set; }
        public int Likes { get; set; }
        public int Downvotes { get; set; }
        public DateTime AddedDate { get; set; }
        public string VidGuardId { get; set; }
        public List<GetCategoryDto> Categories { get; set; }
        public List<MovieTagDto> Tags { get; set; }
    }
}

File: FE/movies-client/src/app/dtos/GetMovieDto.ts
import { GetCategoryDto } from "./GetCategoryDto";
import { TagDto } from "./TagDto";



export class GetMovieDto 
{
    public id!:string;
    public title!:string;
    public description!:string;
    public addedDate!:Date;
    public vidGuardId!:string;
    public views!:number;
    public upvotes!:number;
    public likes!:number;
    public downvotes!:number;
    public posterImageUrl!:string;
    public categories!:Array<GetCategoryDto>;
    public tags!:Array<TagDto>;
}